hsakmt-roct@4.5.0 build fails: CMake: Could NOT find PkgConfig, Could not find UDEV

### Steps to reproduce the issue

`hsakmt-roct@4.5.0` build fails using:
* `spack@develop` (1ea4497802dde0ac63ec9aa95ea4d601331b7e2e from `Thu Jan 13 05:50:11 2022 -0500`
* Ubuntu 20.04, x86_64
* GCC 9.3.0
* CMake 3.22.1

Concrete spec: [hsakmt-roct-7pgzyzz.spec.json.txt](https://github.com/spack/spack/files/7863620/hsakmt-roct-7pgzyzz.spec.json.txt)

Concretization:
```
 -   hsakmt-roct@4.5.0%gcc@9.3.0~ipo+shared build_type=Release arch=linux-ubuntu20.04-x86_64
[+]      ^cmake@3.22.1%gcc@9.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]          ^ncurses@6.2%gcc@9.3.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]              ^pkgconf@1.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]          ^openssl@1.1.1m%gcc@9.3.0~docs certs=system arch=linux-ubuntu20.04-x86_64
[+]              ^perl@5.34.0%gcc@9.3.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]                  ^berkeley-db@18.1.40%gcc@9.3.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu20.04-x86_64
[+]                  ^bzip2@1.0.8%gcc@9.3.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]                      ^diffutils@3.8%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^libiconv@1.16%gcc@9.3.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                  ^gdbm@1.19%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^readline@8.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^zlib@1.2.11%gcc@9.3.0+optimize+pic+shared arch=linux-ubuntu20.04-x86_64
[+]      ^libdrm@2.4.100%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]          ^docbook-xml@4.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]              ^libxml2@2.9.12%gcc@9.3.0~python arch=linux-ubuntu20.04-x86_64
[+]                  ^xz@5.2.5%gcc@9.3.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^docbook-xsl@1.79.2%gcc@9.3.0 patches=a92c39715c54949ba9369add1809527b8f155b7e2a2b2e30cb4b39ee715f2e30 arch=linux-ubuntu20.04-x86_64
[+]          ^libpciaccess@0.16%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]              ^libtool@2.4.6%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^m4@1.4.19%gcc@9.3.0+sigsegv patches=9dc5fbd0d5cb1037ab1e6d0ecc74a30df218d0a94bdd5a02759a97f62daca573,bfdffa7c2eb01021d5849b36972c069693654ad826c1a20b53534009a4ec7a89 arch=linux-ubuntu20.04-x86_64
[+]                      ^libsigsegv@2.13%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]              ^util-macros@1.19.3%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]          ^libpthread-stubs@0.4%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
[+]      ^numactl@2.0.14%gcc@9.3.0 patches=4e1d78cbbb85de625bad28705e748856033eaafab92a66dffd383a3d7e00cc94,62fc8a8bf7665a60e8f4c93ebbd535647cebf74198f7afafec4c085a8825c006,ff37630df599cfabf0740518b91ec8daaf18e8f288b19adaae5364dc1f6b2296 arch=linux-ubuntu20.04-x86_64
[+]          ^autoconf@2.69%gcc@9.3.0 patches=35c449281546376449766f92d49fc121ca50e330e60fefcfc9be2af3253082c2,7793209b33013dc0f81208718c68440c5aae80e7a1c4b8d336e382525af791a7,a49dd5bac3b62daa0ff688ab4d508d71dbd2f4f8d7e2a02321926346161bf3ee arch=linux-ubuntu20.04-x86_64
[+]          ^automake@1.16.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
```

Build error:
```
$> spack mirror add test https://cache.e4s.io/test-workflow
$> spack buildcache keys -it

$> spack install --only dependencies --include-build-deps --cache-only -f ./hsakmt-roct-7pgzyzz.spec.json
... OK

$> spack install --no-cache -f ./hsakmt-roct-7pgzyzz.spec.json
...
==> Installing hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/62/620b39959e0ee5d709b8cf6eb3cc06c8356d72838343756230c638899b10bb9a.tar.gz
==> No patches needed for hsakmt-roct
==> hsakmt-roct: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libdrm-2.4.100-oubvjpazucmha66spq7cpu277ecpl5mh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpthread-stubs-0.4-wngdhpdeuvwyfid6b7rkvbhbm22e4dzr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-e3evcwcbrfcfnv4j2h55y6n6dciu37wi/lib' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-e3evcwcbrfcfnv4j2h55y6n6dciu37wi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libdrm-2.4.100-oubvjpazucmha66spq7cpu277ecpl5mh;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpthread-stubs-0.4-wngdhpdeuvwyfid6b7rkvbhbm22e4dzr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.22.1-uc6y7qdjd6gnm5bkgosu5mj22n7h2jxd' '-DBUILD_SHARED_LIBS:BOOL=ON' '/tmp/root/spack-stage/spack-stage-hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j/spack-src'

3 errors found in build log:
     15    fatal: not a git repository (or any of the parent directories): .git
     16    -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
     17    -- Detected distribution: ubuntu:"20.04"
     18    -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
     19    -- LIBC:/usr/lib/x86_64-linux-gnu/libc.so.6
     20    -- NUMA:/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-e3evcwcbrfcfnv4j2h55y6n6dciu37wi/lib/libnuma.so
  >> 21    CMake Error at /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.22.1-uc6y7qdjd6gnm5bkgosu5mj22n7h2jxd/share/cmake-3.22/Modules/FindPkgConfig.cmake:659
           (message):
     22      pkg-config tool not found
     23    Call Stack (most recent call first):
     24      /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.22.1-uc6y7qdjd6gnm5bkgosu5mj22n7h2jxd/share/cmake-3.22/Modules/FindPkgConfig.cmake:825 (_pkg_check_m
           odules_internal)
     25      CMakeLists.txt:179 (pkg_check_modules)
     26
     27
  >> 28    CMake Error at /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.22.1-uc6y7qdjd6gnm5bkgosu5mj22n7h2jxd/share/cmake-3.22/Modules/FindPkgConfig.cmake:659
           (message):
     29      pkg-config tool not found
     30    Call Stack (most recent call first):
     31      /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.22.1-uc6y7qdjd6gnm5bkgosu5mj22n7h2jxd/share/cmake-3.22/Modules/FindPkgConfig.cmake:825 (_pkg_check_m
           odules_internal)
     32      CMakeLists.txt:180 (pkg_check_modules)
     33
     34
  >> 35    CMake Error at CMakeLists.txt:195 (find_library):
     36      Could not find UDEV using the following names: libudev.so, libudev.a
     37
     38
     39    -- Configuring incomplete, errors occurred!
     40    See also "/tmp/root/spack-stage/spack-stage-hsakmt-roct-4.5.0-7pgzyzzm4mrkezealtm7onag4ekrvh4j/spack-build-7pgzyzz/CMakeFiles/CMakeOutput.log".
```

### Information on your system

* **Spack:** 0.17.1-824-1ea4497802
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[CMakeOutput.log](https://github.com/spack/spack/files/7863669/CMakeOutput.log)
[spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/7863671/spack-build-01-cmake-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7863672/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/7863673/spack-build-out.txt)
[spack-configure-args.txt](https://github.com/spack/spack/files/7863674/spack-configure-args.txt)

@srekolam
@arjun-raj-kuppala

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hsakmt-roct/package.py
+++ b/var/spack/repos/builtin/packages/hsakmt-roct/package.py
@@ -34,6 +34,7 @@ class HsakmtRoct(CMakePackage):
     variant('build_type', default='Release', values=("Release", "Debug", "RelWithDebInfo"), description='CMake build type')
     variant('shared', default=True, description='Build shared or static library')
 
+    depends_on('pkgconfig', type='build', when='@4.5.0:')
     depends_on('cmake@3:', type='build')
     depends_on('numactl')
     depends_on('libdrm', when='@4.5.0:')

